litestar/dto/factory/data_structures.py

```diff
@@ -84,7 +84,7 @@ class FieldDefinition(ParsedParameter):
     """Unique identifier of model that owns the field."""
     default_factory: Callable[[], Any] | None
     """Default factory of the field."""
-    dto_field: DTOField | None
+    dto_field: DTOField
```

Comment:
I'd type it as union `DTOField | Empty`

Reply:
As I mentioned before, this is intentionally not a union with anything else.